internal/bminventory/inventory.go

```diff
@@ -4011,18 +3993,76 @@ func (b *bareMetalInventory) RegisterInfraEnvInternal(
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
-	if err = b.db.Create(&infraEnv).Error; err != nil {
+	if err = tx.Create(&infraEnv).Error; err != nil {
+		log.WithError(err).Error("failed to create infraenv")
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	if err = tx.Commit().Error; err != nil {
+		log.WithError(err).Error("failed to commit transaction registering infraenv")
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	success = true
 	if err = b.GenerateInfraEnvISOInternal(ctx, &infraEnv); err != nil {
 		return nil, err
 	}
 
-	success = true
 	return b.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *infraEnv.ID})
 }
 
+func (b *bareMetalInventory) validateInfraEnvCreateParams(ctx context.Context, params installer.RegisterInfraEnvParams, cluster *common.Cluster) error {
+	var err error
+	if cluster != nil && cluster.CPUArchitecture != "" && cluster.CPUArchitecture != params.InfraenvCreateParams.CPUArchitecture {
```

Comment:
This looks new to me.
Why add it now?

Reply:
It was originally here: https://github.com/openshift/assisted-service/pull/3791/files/43514d7cfceae56afb0bd6ea465c5ecd7c37af16#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cL4075-L4077

(Does this link work?)

Reply:
No, but I found where it was. Makes sense.